config/module.go

```diff
@@ -42,7 +42,7 @@ type Module struct {
 	Environment map[string]string `json:"env,omitempty"`
 
 	// FirstRunTimeout is the timeout duration for the first run script.
-	FirstRunTimeout time.Duration `json:"first_run_timeout,omitempty"`
+	FirstRunTimeout goutils.Duration `json:"first_run_timeout,omitempty"`
```

Comment:
What does a timeout of `0s` mean? Looks like you treat it as "use default" and not "infinite"? Seems fine to me, but maybe document here.

Reply:
oh yeah that's a good catch, will document this explicitly for sure

Reply:
@benjirewis is a timeout of `0s` actually infinite or immediate? my understanding is that it's immediate. this doesn't change your suggestion for documentation - I'm still going to do that - but just want to confirm my understanding.

Reply:
confirmed for myself, seems like `0s` or less is immediate [per my testing](https://goplay.tools/snippet/AMF5jPG2sxc).

Reply:
let me know if this doc works for you: https://github.com/viamrobotics/rdk/pull/4511/commits/03b6aa2fa23041d71e3de5492a61dc1baf239ee3

Reply:
Seems good to me! Thanks.

> is a timeout of 0s actually infinite or immediate?

I guess in a formal sense: a timeout of 0s should put a deadline on the context of `time.Now()`, which would mean that attempting to use the context would almost certainly result in a `context deadline exceeded` error, as you would not be at `time.Now() + epsilon` where `epsilon` is some non-zero positive duration.